llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -16890,3 +16889,13 @@ SITargetLowering::lowerIdempotentRMWIntoFencedLoad(AtomicRMWInst *AI) const {
   AI->eraseFromParent();
   return LI;
 }
+
+bool SITargetLowering::hasAndNot(SDValue Op) const {
+  // Return false if the operation is divergent, as AND-NOT is a scalar-only
+  // instruction.
+  if (Op->isDivergent() || !Op->isMachineOpcode())
+    return false;
+
+  EVT VT = Op.getValueType();
+  return VT == MVT::i32 || VT == MVT::i64;
+}
```

Comment:
@arsenm, just checking for i32 & i64 is sufficient on hardware behalf to say it has AndNot support?

Reply:
Hi Vikash, I think AMDGPU has supported the AndNot instruction for a long time because I noticed there have been many lit tests for it. This PR just aims to implement hasAndNot. :)